### PR TITLE
Update SQLite to the 3.27.2 release

### PR DESCRIPTION
**Pull request by mistachkin**
These changes are designed to update the included SQLite source code to the latest pre-3.27 from upstream, while maintaining the necessary comdb2 customizations.

### CODE REVIEW

**Comment by riverszhang89** [comment]
The error was likely caused by OS upgrade on BTDA boxes. I've resubmitted your pull request.

File: sqlite/src/wherecode.c [hunk 1]
```diff
@@ -1836,8 +1836,13 @@ Bitmask sqlite3WhereCodeOneLoopStart(
     ** the cursor. In this case it is important to do the full evaluation,
     ** as the result of the expression may not be NULL, even if all table
     ** column values are.  https://www.sqlite.org/src/info/7fa8049685b50b5a
+    **
+    ** Also, do not do this when processing one index an a multi-index
```
Comment (nirbhayc) [inline sqlite/src/wherecode.c]: typo "one index on a multi-index OR clause"

Comment (mistachkin) [reply sqlite/src/wherecode.c]: I'll have to fix this upstream to avoid merge conflicts.

**Comment by markhannum** [comment]
Approved pending ROBOMARK :)

**Comment by mistachkin** [comment]
> Looks good to me. We discussed optimizing by not running prepare() twice, but that can go in a diff PR if too much work.

I've made the necessary changes to optimize the normalization code.  We no longer prepare twice.

**Pull request merged**